src/wasm/wasm-validator.cpp

```diff
@@ -2059,6 +2111,21 @@ void FunctionValidator::visitTry(Try* curr) {
   shouldBeTrue(curr->catchBodies.size() - curr->catchEvents.size() <= 1,
                curr,
                "the number of catch blocks and events do not match");
+
+  shouldBeFalse(curr->isCatch() && curr->isDelegate(),
+                curr,
+                "try cannot have both catch and delegate at the same time");
+  shouldBeTrue(curr->isCatch() || curr->isDelegate(),
+               curr,
+               "try should have either catches or a delegate");
+
+  // (try $l (delegate $l)) is invalid. So we pop the current try's name before
+  // checking the delegate target.
+  delegateTargetNames.erase(curr->name);
```

Comment:
is this needed? I think it will have already been erased in `visitPreCatch`.

Reply:
I think you're right. Removed.